### PR TITLE
[ML] Inference API filtering out rate limit service settings field

### PR DESCRIPTION
**Pull request by jonathan-buttner**
This PR hides the `rate_limit` field since it isn't used yet.

I refactored some of the code since about all the 3rd party services need the same boilerplate method. So I moved it into an abstract class.

### CODE REVIEW

File: x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/cohere/rerank/CohereRerankServiceSettingsTests.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+    public void testToXContent_WritesAllValues() throws IOException {
+        var serviceSettings = new CohereRerankServiceSettings(
+            new CohereServiceSettings("url", SimilarityMeasure.COSINE, 5, 10, "model_id", new RateLimitSettings(3))
+        );
+
+        XContentBuilder builder = XContentFactory.contentBuilder(XContentType.JSON);
+        serviceSettings.toXContent(builder, null);
+        String xContentResult = Strings.toString(builder);
+        // TODO we probably shouldn't allow configuring these fields for reranking
```
Comment (maxhniebergall) [inline x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/cohere/rerank/CohereRerankServiceSettingsTests.java:48]: Thanks for noticing this! If you haven't already, I'll create an issue for this.

Comment (jonathan-buttner) [reply x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/cohere/rerank/CohereRerankServiceSettingsTests.java:48]: Yeah no worries! I created one here: https://github.com/elastic/elasticsearch/issues/108260

**Review by maxhniebergall — APPROVED** [review]
LGTM

The refactoring wtih FilterXContent is nice, and I appreciate the clear naming of xContent fragements and objects.

**Review by davidkyle — APPROVED** [review]
LGTM

**Pull request merged**